test/topology_experimental_raft/test_fencing.py

```diff
@@ -0,0 +1,176 @@
+#
+# Copyright (C) 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+from test.pylib.random_tables import RandomTables, Column, IntType, CounterType
+from test.pylib.util import unique_name, wait_for_cql_and_get_hosts, wait_for
+from cassandra import WriteFailure, ConsistencyLevel
+from test.pylib.internal_types import ServerInfo
+from test.pylib.rest_client import ScyllaMetrics
+from cassandra.pool import Host # type: ignore # pylint: disable=no-name-in-module
+from cassandra.query import SimpleStatement
+from test.topology.conftest import skip_mode
+import pytest
+import logging
+import time
+
+
+logger = logging.getLogger(__name__)
+
+
+def host_by_server(hosts: list[Host], srv: ServerInfo):
+    for h in hosts:
+        if h.address == srv.ip_addr:
+            return h
+    raise ValueError(f"can't find host for server {srv}")
+
+
+async def set_version(manager: ManagerClient, host: Host, new_version: int):
+    await manager.cql.run_async("update system.topology set version=%s where key = 'topology'",
+                                parameters=[new_version],
+                                host=host)
+
+
+async def set_fence_version(manager: ManagerClient, host: Host, new_version: int):
+    await manager.cql.run_async("update system.scylla_local set value=%s where key = 'topology_fence_version'",
+                                parameters=[str(new_version)],
+                                host=host)
+
+
+async def get_version(manager: ManagerClient, host: Host):
+    rows = await manager.cql.run_async(
+        "select version from system.topology where key = 'topology'",
+        host=host)
+    return rows[0].version
+
+
+def send_errors_metric(metrics: ScyllaMetrics):
+    return metrics.get('scylla_hints_manager_send_errors')
+
+
+def sent_metric(metrics: ScyllaMetrics):
+    return metrics.get('scylla_hints_manager_sent')
+
+
+@pytest.mark.asyncio
+async def test_fence_writes(request, manager: ManagerClient):
+    logger.info("Bootstrapping first two nodes")
+    servers = [await manager.server_add(), await manager.server_add()]
```

Comment:
For some reason, if I change this to `servers = await asyncio.gather(manager.server_add(), manager.server_add())` the driver sometimes spews `12:39:59.014 INFO> Driver hasn't yet learned about hosts: {'127.157.193.8'}` on me, sometimes not. This happens on `host2 = (await wait_for_cql_and_get_hosts(manager.cql, [servers[2]], time.time() + 60))[0]`, this wait function fails. @kbr-scylla @alecco maybe you came across this, is it a known problem?

Reply:
Nope, that's weird

Reply:
https://github.com/scylladb/scylladb/issues/15119